### PR TITLE
Task/calm 19390 task library support

### PR DESCRIPTION
**Pull request by kumarsarath588**
Calm DSL support for seeding task-library items.
Sample commands:
calm get library tasks
calm create library task -f SampleTask.py
calm import library task -f ~/Calm/blueprints/library/task-library/Kubernetes/Etcd_Install_and_Configure.sh
calm describe library task Etcd_Install_and_Configure
calm delete library task Etcd_Install_and_Configure

### CODE REVIEW

File: calm/dsl/cli/library_tasks_commands.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+(-f | --file) supports:\n
+\t.json     - Full json payload download from Calm API (v3 #GET) or using `calm describe <task_name> -o json`\n
+\t.sh       - Shell script file\n
+\t.escript  - Escript file\n
+\t.ps1      - Powershell Script File\n
+
+Note: HTTP tasks is supported only from downloaded .json.
+
+Sample:\n
```
Comment (abhijeetkaurav1st) [inline calm/dsl/cli/library_tasks_commands.py:108]: Pls mention how can we create task of various type `EXECUTE / SET VARIABLE / HTTP` clearly.
Better if we can provide some sample examples.

File: calm/dsl/cli/library_tasks.py [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+    task = get_task(client, task_name, all=True)
+
+    res, err = client.task.read(task["metadata"]["uuid"])
+    if err:
+        raise Exception("[{}] - {}".format(err["code"], err["error"]))
+
+    task = res.json()
+
+    if out == "json":
+        task.pop("status", None)
```
Comment (abhijeetkaurav1st) [inline calm/dsl/cli/library_tasks.py:118]: Pls print status dict also in output.

File: calm/dsl/builtins/models/tasks.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from .entity import EntityType
+
+
+# Task
```
Comment (abhijeetkaurav1st) [inline calm/dsl/builtins/models/tasks.py:4]: Any need of this file ?

Comment (abhijeetkaurav1st) [reply calm/dsl/builtins/models/tasks.py:4]: If yes.
These are same task that are defined in `calm/dsl/builtins/models/task.py`.
Try to add your logic somewhat there.

File: calm/dsl/api/task.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from .resource import ResourceAPI
+
+
+class TaskAPI(ResourceAPI):
```
Comment (abhijeetkaurav1st) [inline calm/dsl/api/task.py:4]: Can we change the name to some `TaskLibraryApi` as they are used only in task library ?
(Maybe I am not correct enough.)

File: calm/dsl/cli/library_tasks_commands.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    "--description", "-d", default=None, help="Blueprint description (Optional)"
+)
+@click.option(
+    "--force",
+    "-fc",
+    is_flag=True,
+    default=False,
+    help="Updates existing task library item with the same name.",
+)
+def _create_task(task_file, name, description, force):
```
[the remark this answers is not in the document]
Comment (kumarsarath588) [reply calm/dsl/cli/library_tasks_commands.py:96]: User has to mention task type in blueprint only, and import these tasks. So we don't have task type only script type.

File: calm/dsl/cli/library_tasks_commands.py (hunk 1 above)
Comment (kumarsarath588) [reply calm/dsl/cli/library_tasks_commands.py:108]: calm create library task --name=HTTPGetVM -f HTTPGetVM.json -> HTTP task type
calm create library task --name="Install IIS" -f Install_IIS.ps1 -> Execute task type
calm create library task -f Install_Docker.sh -> Execute task type
there is no set variable that distinction should be done in BP only.

File: calm/dsl/cli/library_tasks.py (hunk 2 above)
Comment (kumarsarath588) [reply calm/dsl/cli/library_tasks.py:118]: done

File: calm/dsl/builtins/models/tasks.py (hunk 3 above)
Comment (kumarsarath588) [reply calm/dsl/builtins/models/tasks.py:4]: removed the file as its not required

File: calm/dsl/api/task.py (hunk 4 above)
Comment (kumarsarath588) [reply calm/dsl/api/task.py:4]: done changed to TaskLibraryApi

**Review by abhijeetkaurav1st — APPROVED** [review]

**Pull request merged**